db/background_mgr.go

```diff
@@ -0,0 +1,200 @@
+//  Copyright 2012-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package db
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/couchbase/sync_gateway/base"
+)
+
+type BackgroundProcessState string
+
+const (
+	BackgroundProcessStateRunning  BackgroundProcessState = "running"
+	BackgroundProcessStateStopped  BackgroundProcessState = "stopped"
+	BackgroundProcessStateStopping BackgroundProcessState = "stopping"
+	BackgroundProcessStateError    BackgroundProcessState = "error"
+)
+
+type BackgroundProcessAction string
+
+const (
+	BackgroundProcessActionStart BackgroundProcessAction = "start"
+	BackgroundProcessActionStop  BackgroundProcessAction = "stop"
+)
+
+// BackgroundManagerI this is an interface satisfied by BackgroundManager
+// This is technically not needed, however, it allows easily visibility into exposed methods
```

Comment:
I feel like the normal definition of public methods already does a sufficient job of providing visibility to exposed methods, and we should remove this interface if it's not used.

Reply:
Also it's not accurate, since SetRunState/GetRunState/SetError are public.